platforms/posix/src/main.cpp

```diff
@@ -578,26 +578,30 @@ void print_usage()
 bool is_already_running(int instance)
 {
 	const std::string file_lock_path = std::string(LOCK_FILE_PATH) + '-' + std::to_string(instance);
-	struct flock fl;
 	int fd = open(file_lock_path.c_str(), O_RDWR | O_CREAT, 0666);
 
 	if (fd < 0) {
+		close(fd);
```

Comment:
This doesn't make sense, as fd is invalid here. Please take a closer look at the coverity output, the leak is happening further down (for all other returns).

Reply:
I initially agreed with your assessment, and I had originally added this at the bottom of the method until I saw that caused the CI tools to fail.  After digging a bit more, I added this here for two reasons: 
1) From the Open Group documentation on POSIX close():
>Closing a File Descriptor
>
>In the following example, close() is used to close a file descriptor after an unsuccessful attempt is made to associate that file descriptor with a stream.
>
>#include <stdio.h>
>#include <unistd.h>
>#include <stdlib.h>
>
>
>#define LOCKFILE "/etc/ptmp"
>...
>int pfd;
>FILE *fpfd;
>...
>if ((fpfd = fdopen (pfd, "w")) == NULL) {
>    close(pfd);
>    unlink(LOCKFILE);
>    exit(1);
>}
2) CI tools pass with this usage.

I'm honestly interested in learning what the right thing to do about this is.  Please let me know your thoughts!

Reply:
> From the Open Group documentation on POSIX close():

Looking at that example ([here](https://pubs.opengroup.org/onlinepubs/009695399/functions/close.html)): `close` is called on `pfd`, which at that point is a valid file descriptor (the example is not complete). The call that failed in that example is `fdopen`, not `open`.
That page also shows you that you just get an `EBADF` (The fildes argument is not a valid file descriptor.) when attempting to close an invalid file descriptor. So please don't do it.

> CI tools pass with this usage.

You need to be careful here. A passing CI does not mean it takes that branch, or that it's the correct thing to do. CI helps but local testing is still required.

Reply:
Thanks for the feedback @bkueng .  This PR seemed like low hanging fruit to begin with, however now it appears not to be. :}  [Here](https://en.wikipedia.org/wiki/Open_(system_call)) was my other rational for the previous close() call placement (not relevant with the current commit):
>If the file is being created, the filesystem may allocate the default initial amount of storage or a specified amount depending on the file system capabilities. If this fails an error will be returned. Updating the directory with the new entry **may be performed or it may be delayed until the close** is performed.

(emphasis added)

With local hardware testing on the bench I don't see the failures that the CI tools show, (I'm not on a mac).  All of the hardware tests show the same with this PR on my hardware as they do with current master, running `tests all`:
![image](https://user-images.githubusercontent.com/2497951/54625523-12c2f700-4a35-11e9-84d1-05192939e339.png)

I'm at a loss.  Let me know if I should just close this PR.

Reply:
I just looked at the context: this is used for checking if an instance is already running and the file is opened with `O_CREAT`. This fd has to be kept open until the application terminates (which explains the malfunctions you see), meaning this is a false positive from coverity. You can mark to ignore it.

The gist here is: don't trust coverity's output too much :)

I'm closing this PR, I really should have seen this right away. Sorry for the troubles.